web-ng/src/app/components/project-list/project-list.component.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, OnDestroy, ElementRef, OnInit } from '@angular/core';
+
+import { ProjectService } from '../../services/project/project.service';
+import { Subscription } from 'rxjs';
+import { MatDialog } from '@angular/material/dialog';
+import { UserProfilePopupComponent } from '../user-profile-popup/user-profile-popup.component';
+import { Project } from '../../shared/models/project.model';
+import { NavigationService } from '../../services/navigation/navigation.service';
+
+@Component({
+  selector: 'app-project-list',
+  templateUrl: './project-list.component.html',
+  styleUrls: ['./project-list.component.scss'],
+})
+export class ProjectListComponent implements OnInit, OnDestroy {
+  projects?: Project[];
```

Comment:
Please use an immutable List here and throughout instead of an array. In the reactive model, items emitted from a stream should not be mutable or there could be surprise side-effects if code inadvertently modifies the contained value.

Reply:
updated.